src/builder.py

```diff
@@ -41,7 +44,7 @@ def __exit__(self, type, value, traceback):
 
 
 class FileHandle:
-    def __init__(self, builder, path):
+    def __init__(self, builder: 'Builder', path: str) -> None:
```

Comment:
I have to confess I'm not an expert on Python's type hinting: what's with the string quoted type name?

Reply:
It's a [kludge for handling forward references](https://www.python.org/dev/peps/pep-0484/#forward-references).